core/services/aws_service.rb

```diff
@@ -293,6 +293,70 @@ def machine_types_list(supported_types = nil)
     end
   end
 
+  # Searches for instance types that match the image requirements
+  # @param architecture [String] CPU architecture type
+  # @param virtualization_type [String] AMI virtualization type
+  # @return [Array<String>] instance types
+  def machine_types_list_from_requirements(architecture, virtualization_type)
+    types = []
+    @client.get_instance_types_from_instance_requirements(
+      {
+        architecture_types: [architecture],
+        virtualization_types: [virtualization_type],
+        instance_requirements: {
+          v_cpu_count: { min: 1 },
+          memory_mi_b: { min: 512 }
+        }
+      }
+    ).each do |response|
+      types += response.instance_types.to_a
+    end
+    types.map(&:instance_type)
+  end
+
+  # Searches for instance types in the configured Availability Zone
+  # @return [Array<String>] instance types
+  def machine_types_list_from_zone
+    zone = @aws_config['availability_zone']
+    types = []
+    @client.describe_instance_type_offerings(
+      {
+        location_type: 'availability-zone',
+        filters: [{ name: 'location', values: [zone] }]
+      }
+    ).each do |response|
+      types += response.instance_type_offerings.to_a
+    end
+    types.map(&:instance_type)
+  end
+
+  # Searches for an image by AMI and obtains required CPU architecture and virtualization type
+  # @param ami [String] Amazon Machine Image id
+  # @return [Hash] image parameters in format { architecture, virtualization_type }.
+  def get_ami_parameters(ami)
```

Comment:
@amvasilyev, should we make added methods (except `supported_instance_types`) private, as they aren't used outside the class?

Reply:
@GrigoriyKostryukov yes. We will make it public once it will be required by other modules.

Also consider to make a single wrapper method, so the user would just call it, not two different methods.